File: JAVA/DprMain.java

import java.util.Scanner;
import java.util.ArrayList;

public class DprMain {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        ArrayList<Dpr> list = new ArrayList<>();

        int a = 0;
        do {
            System.out.println("Pilihan proses : \n");
            System.out.println("1. Tambah\n 2. Hapus\n 3. Sunting\n 4. Selesai\n");
            a = sc.nextInt();

            switch (a) {
                case 1:
                    System.out.print("Masukkan ID, Nama, Bidang, Partai: ");
                    String id = sc.next();
                    String name = sc.next();
                    String bidang = sc.next();
                    String partai = sc.next();
                    Dpr temp1 = new Dpr();
                    temp1.setId(id); temp1.setName(name); temp1.setBidang(bidang); temp1.setPartai(partai);
                    list.add(temp1);
                    break;
                case 2:
                    System.out.print("Masukkan ID anggota yang ingin dihapus: ");
                    String idHapus = sc.next();
                    for (int o = list.size() - 1; o >= 0; o--) {
                        if (list.get(o).getId().equals(idHapus)) {
                            list.remove(o);
                        }
                    }
                    break;
                case 3:
                    System.out.print("Masukkan ID anggota yang ingin disunting: ");
                    String idSunting = sc.next();
                    for (int j = 0; j < list.size(); j++) {
                        if (list.get(j).getId().equals(idSunting)) {
                            System.out.print("Masukkan Nama, Bidang, Partai baru: ");
                            name = sc.next();
                            bidang = sc.next();
                            partai = sc.next();
                            Dpr temp2 = new Dpr();
                            temp2.setId(idSunting);temp2.setName(name); temp2.setBidang(bidang); temp2.setPartai(partai);
                            list.set(j, temp2);
                            break;
                        }
                    }
                    break;
                case 4:
                    break;
                default:
                    System.out.println("Pilihan tidak valid.");
            }

            System.out.println("Anggota DPR : ");
            System.out.println("+-----+------------+----------+------------+");
            System.out.println("| ID  | Nama       | Bidang   | Partai     |");
            System.out.println("+-----+------------+----------+------------+");
            for (int i = 0; i < list.size(); i++) {
                System.out.printf("| %-3s | %-10s | %-8s | %-10s |%n",
                        list.get(i).getId(), list.get(i).getName(), list.get(i).getBidang(), list.get(i).getPartai());
            }
            System.out.println("+-----+------------+----------+------------+");
        } while (a != 4);

        sc.close();
    }
}

File: PYTHON/DprMain.py
from Dpr import Dpr


#array
list = []

#iterasi simple
n = int(input())
for i in range (n):
    id = str(input())
    name = str(input())
    bidang = str(input())
    partai = str(input())
    #gender = input()[0]
    
    list.append(Dpr(id, name, bidang, partai))

#foreach
header = "| ID        | Nama              | bidang           | Partai        |"
print("-" * len(header))
print(header)
print("-" * len(header))
i = 0
for anggota in list:
    print("| {:<10} | {:<17} | {:<15} | {:<12} |".format(anggota.get_id(), anggota.get_name(), anggota.get_bidang(), anggota.get_partai()))
    print("-" * len(header))
    #i += 1

perintah = 0
while perintah != 4:
    print("Pilihan proses :\n 1. Tambah\n 2. Hapus\n 3. Sunting\n 4. Selesai\n")
    perintah = int(input())
    if perintah == 1:           #TAMBAH
        
        id = input()
        name = input()
        bidang = input()
        partai = input()
        list.append(Dpr(id, name, bidang, partai))
        
        header = "| ID        | Nama              | bidang           | Partai        |"
        print("-" * len(header))
        print(header)
        print("-" * len(header))
        i = 0
        for anggota in list:
            print("| {:<10} | {:<17} | {:<15} | {:<12} |".format(anggota.get_id(), anggota.get_name(), anggota.get_bidang(), anggota.get_partai()))
            print("-" * len(header))
    
    elif perintah == 2:     #HAPUS
        id = input()
        list = [anggota for anggota in list if anggota.get_id() != id]
        header = "| ID        | Nama              | bidang           | Partai        |"
        print("-" * len(header))
        print(header)
        print("-" * len(header))
        i = 0
        for anggota in list:
            print("| {:<10} | {:<17} | {:<15} | {:<12} |".format(anggota.get_id(), anggota.get_name(), anggota.get_bidang(), anggota.get_partai()))
            print("-" * len(header))
            
    elif perintah == 3:     #EDIT
        id = input()
        name = input()
        bidang = input()
        partai = input()
        
        for i, anggota in enumerate(list):
            if anggota.get_id() == id:
                list[i].set_name(name)
                list[i].set_bidang(bidang)
                list[i].set_partai(partai)
                break
            
        header = "| ID        | Nama              | bidang           | Partai        |"
        print("-" * len(header))
        print(header)
        print("-" * len(header))
        i = 0
        for anggota in list:
            print("| {:<10} | {:<17} | {:<15} | {:<12} |".format(anggota.get_id(), anggota.get_name(), anggota.get_bidang(), anggota.get_partai()))
            print("-" * len(header))
    